run/test/specs/locators/index.ts

```diff
@@ -0,0 +1,93 @@
+import { DeviceWrapper } from "../../../types/DeviceWrapper";
+import { StrategyExtractionObj } from "../../../types/testing";
+import { SupportedPlatformsType } from "../utils/open_app";
+
+export abstract class LocatorsInterface {
+  protected readonly platform: SupportedPlatformsType;
+
+  abstract build(): StrategyExtractionObj;
+
+  constructor(device: DeviceWrapper) {
+    if (device.isAndroid()) {
+      this.platform = "android";
+    } else if (device.isIOS()) {
+      this.platform = "ios";
+    } else {
+      console.warn("unsupported device type:", device);
+      throw new Error("unsupported device type");
+    }
+  }
+
+  protected isIos() {
+    return this.platform === "ios";
+  }
+
+  protected isAndroid() {
+    return this.platform === "android";
+  }
+}
+// When applying a nickname or username change
+export class TickButton extends LocatorsInterface {
+  public build() {
+    if (this.isIos()) {
+      return { strategy: "accessibility id", selector: "Done" } as const;
+    }
+    return { strategy: "accessibility id", selector: "Apply" } as const;
+  }
```

Comment:
it might be better to have a switch instead of a if/else, just because we might at some point add a desktop type and it would be nice to not blindly use the `else` case for it.
```suggestion
    switch(this.platform){
       case "android": return { strategy: "accessibility id", selector: "Apply" } as const;
       case "ios": return { strategy: "accessibility id", selector: "Done" } as const;
    }
```

What do you think?

Reply:
yeah i agree, its much nicer, will update now